pkg/openwsn/sock/openwsn_sock_udp.c

```diff
@@ -0,0 +1,556 @@
+            if (res > 0) {
+                /* copy received data */
+                memcpy(data, pkt, res);
+                /* free packet */
+                tmp = sock_udp_recv_buf_aux(sock, &pkt, &ctx, timeout, remote,
+                                            aux);
+                assert(tmp == 0);
+            }
+            (void) tmp;
+        }
+        else {
+            res = -ENOBUFS;
+            /* free packet */
+            tmp = sock_udp_recv_buf_aux(sock, &pkt, &ctx, timeout, remote, aux);
+            assert(tmp == 0);
+        }
+    }
+    return res;
+}
+
+ssize_t sock_udp_recv_buf_aux(sock_udp_t *sock, void **data, void **buf_ctx,
+                              uint32_t timeout, sock_udp_ep_t *remote,
+                              sock_udp_aux_rx_t *aux)
+{
+    (void) aux;
+    sock_udp_ep_t ep;
+    OpenQueueEntry_t *pkt = NULL;
+    msg_t msg;
+
+    assert((sock != NULL) && (data != NULL) && (buf_ctx != NULL));
+
+    if (*buf_ctx != NULL) {
+        *data = NULL;
+        openqueue_freePacketBuffer(*buf_ctx);
+        *buf_ctx = NULL;
+        return 0;
+    }
+    if (sock->gen_sock.local.family == AF_UNSPEC) {
+        return -EADDRNOTAVAIL;
+    }
+
+#ifdef MODULE_ZTIMER_USEC
+    ztimer_t timer;
+
+    if ((timeout != SOCK_NO_TIMEOUT) && (timeout != 0)) {
+        timer.callback = _timeout_cb;
+        timer.arg = &sock->mbox;
+        ztimer_set(ZTIMER_USEC, &timer, timeout);
+    }
+#endif /* MODULE_ZTIMER_USEC */
+    if (timeout != 0) {
+        mbox_get(&sock->mbox, &msg);
+    }
+    else {
+        if (!mbox_try_get(&sock->mbox, &msg)) {
+            return -EAGAIN;
+        }
+    }
+#ifdef MODULE_ZTIMER_USEC
+    if ((timeout != SOCK_NO_TIMEOUT) && (timeout != 0)) {
+        ztimer_remove(ZTIMER_USEC, &timer);
+    }
+#endif /* MODULE_ZTIMER_USEC */
+    switch (msg.type) {
+    case _MSG_TYPE_RECV_PKT:
+        pkt = msg.content.ptr;
+        break;
+#ifdef MODULE_ZTIMER_USEC
+    case _TIMEOUT_MSG_TYPE:
+        if (msg.content.value == _TIMEOUT_MAGIC) {
+            return -ETIMEDOUT;
+        }
+#endif /* MODULE_ZTIMER_USEC */
+    /* Falls Through. */
+    default:
+        return -EINVAL;
+    }
+
+    if (pkt == NULL) {
+        return -EINVAL;
+    }
+
+    /* copy source port and address to end point */
+    ep.family = AF_INET6;
+    ep.port = pkt->l4_sourcePortORicmpv6Type;
+    ep.netif = CONFIG_OPENWSN_NETIF_ID;
+    memcpy(&ep.addr, pkt->l3_sourceAdd.addr_128b, LENGTH_ADDR128b);
+
+    if (remote != NULL) {
+        /* return remote if requested */
+        memcpy(remote, &ep, sizeof(sock_udp_ep_t));
+    }
+
+    if ((sock->gen_sock.remote.family != AF_UNSPEC) &&  /* check remote end-point if set */
+        ((sock->gen_sock.remote.port != pkt->l4_sourcePortORicmpv6Type) ||
+                                                        /* We only have IPv6 for now, so just comparing the whole end point
+                                                         * should suffice */
```

Comment:
Line-length. Maybe put the comments above the condition you are commenting on? This is the usual approach in the remaining code base.

Reply:
applied